apps/system/js/notifications.js

```diff
@@ -426,9 +430,9 @@ var NotificationScreen = {
     notificationNode.dataset.type = type;
     notificationNode.dataset.manifestURL = manifestURL;
 
-    if (detail.icon) {
+    if (detail.appIcon) {
```

Comment:
@Ashish-Developer it's strange, because when I made changes for homescreen notifications both values ``detail.icon`` & ``detail.appIcon`` were populated at least for missed calls and incoming messages.